aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+
+                    # Handle non-chunk dimensions (specific indices / ints)
+                    else:
+                        # Store the dim for later to inform to use the provided index
+                        use_selected_or_np_map[dim] = index_op
+                        retrieve_shape.append(1)
+
+            # Create list of planes that we will add each plane to, later we reshape
+            # Create empty arr with the desired shape to enumerate over the np index
+            planes: List[np.ndarray] = []
+            np_array_for_indices = np.empty(tuple(retrieve_shape), dtype=object)
+            for np_index, _ in np.ndenumerate(np_array_for_indices):
+                # Get each plane's index selection operations
+                # If the dimension is None, use the enumerated np index
+                # If the dimension is not None, use the provided value
+                plane_indices: Dict[str, int] = {"S": scene}
+
+                # Handle MosaicTile
+                if DimensionNames.MosaicTile in use_selected_or_np_map:
+                    if use_selected_or_np_map[DimensionNames.MosaicTile] is None:
+                        plane_indices["M"] = np_index[
+                            retrieve_dims.index(DimensionNames.MosaicTile)
+                        ]
+                    else:
+                        plane_indices["M"] = use_selected_or_np_map[  # type: ignore
+                            DimensionNames.MosaicTile
+                        ]
+
+                # Handle Time
+                if DimensionNames.Time in use_selected_or_np_map:
+                    if (
+                        DimensionNames.Time in retrieve_dims
+                        and use_selected_or_np_map[DimensionNames.Time] is None
+                    ):
+                        plane_indices["T"] = np_index[
+                            retrieve_dims.index(DimensionNames.Time)
+                        ]
+                    else:
+                        plane_indices["T"] = use_selected_or_np_map[  # type: ignore
+                            DimensionNames.Time
+                        ]
+
+                # Handle Channels
+                if DimensionNames.Channel in use_selected_or_np_map:
+                    if (
+                        DimensionNames.Channel in retrieve_dims
+                        and use_selected_or_np_map[DimensionNames.Channel] is None
+                    ):
+                        plane_indices["C"] = np_index[
+                            retrieve_dims.index(DimensionNames.Channel)
+                        ]
+                    else:
+                        plane_indices["C"] = use_selected_or_np_map[  # type: ignore
+                            DimensionNames.Channel
+                        ]
+
+                # Handle SpatialZ
+                if DimensionNames.SpatialZ in use_selected_or_np_map:
+                    if use_selected_or_np_map[DimensionNames.SpatialZ] is None:
```

Comment:
Same question here RE `DimensionNames.SpatialZ in retrieve_dims` - is that needed? If so maybe this could be a common function? These ops all look quite similar